src/nrsc5.c

```diff
@@ -69,7 +73,7 @@ static int do_auto_gain(nrsc5_t *st)
 
         uint8_t max_sample = 0;
         uint8_t min_sample = 255;
-        for (int j = 0; j < len; j++)
+        for (int j = len/4; j < len; j++)
```

Comment:
I had to change this to ignore the samples towards the beginning of the buffer, since they are leftover from the "old" gain value. This wasn't an issue when we tested gains in increasing order, but the binary search adjusts the gain in both directions.